xsl/gnc-custom-chm.xsl

```diff
@@ -20,6 +20,26 @@
 book      nop
 </xsl:param>
 
+<!-- language specific character code for traditional Windows codepages
+     ja: Shift_JIS
+     ru: KOI8-R
+-->
+<xsl:param name="htmlhelp.encoding">
+  <xsl:choose>
+    <xsl:when test="$gnc.lang = 'ja'">Shift_JIS</xsl:when>
+    <xsl:when test="$gnc.lang = 'ru'">KOI8-R</xsl:when>
+    <xsl:otherwise>ISO-8859-1</xsl:otherwise>
```

Comment:
If UTF-8 is really no option we need ISO 8859-15 for Americas, Western Europe, Oceania, and much of Africa
Otherwise we have no currency symbol.
[ISO/IEC 8859-15](https://en.wikipedia.org/wiki/ISO/IEC_8859-15)

Reply:
Windows code pages are not compliant with ISO codes like ISO-8859-*. Moreover, the multibyte charset is more complex. 
Windows-1252, CP1252, MSCP1252, and WINDOWS-1252 are not suitable for iexplore.exe.

So we should not change it unless problems don't happen.

See comments of files and directories below: 
https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=tree;f=lib,
https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=tree;f=libcharset/lib,
https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=blob;f=libcharset/lib/localcharset.c.

Traditional charsets are very complex.
---
wantfix

Reply:
We have a problem:
ISO-8859-1 has no `€` symbol, which is the main currency symbol in our docs in de, it, pt. It is also used in C.
That was the main reason to replace ISO-8859-1 by ISO-8859-15.

Reply:
Ideally, you're right.
In practice, you haven't checked the output of the file.

Euro sign is coded by xsltproc to &#8364; in ISO-8859-1. And iexplore.exe decode and render it as Euro sign correctly. Therefore no problem will happen.

Moreover, HORIZONTAL_ELLIPSIS &#8230; is not rendered on TOC of htmlhelp.exe.

I think it isn't very sensible to stick to standards for the legacy CHM files. Stick to their outputs. Of course, it is NOT ideal. 

![de-guide-ch03](https://user-images.githubusercontent.com/85024274/135708473-89a1f62d-f864-41af-8539-671685876731.png)

Reply:
![de-guide-ch07](https://user-images.githubusercontent.com/85024274/135708780-5a85f620-94f9-4374-a302-9bb496fd57da.png)

Reply:
Another option is to stop using CHM. If we use HTML files, you will not bother to character sets. We can directly use UTF-8.

Reply:
7 byte instead of one. ;-)
I do not understand why you insists on using the obsolete ISO-8859-1 and spent days on research instead of changing it simply to the more recent ISO-8859-15.

Usually specific help browsers have some benefits over normal web browsers like extended search  facilities. As I am no Windows user, I have no experience and no strong opinion. As the currently used browser is no longer supported by MS, what are they offering instead?

Perhaps you should collect the opinion of Windows users over the mailing list.

Reply:
From my point of view:
* ISO-8859-1 is the default encoding of xsltproc. And no one noticed problems over 10 years. Actually, no problem happens except Japanese and Russian.
* You requested me to change the encoding to UTF-8. I answered I can't expect what will happen in the TOC pane.
* You requested me to use Windows-1252. I answered it is not suitable for HTML. In fact, both documents and TOC are formatted with HTML internally for CHM.
* You requested me to use ISO-8859-15 because the Euro sign is not defined in ISO-8859-1. But the Euro sign is rendered in the HTML pane correctly. And I found you haven't either viewed or tested the output of CHM.

I beg you to view, test, and compare the files before requesting me to change the settings.
Your requests are not fit the reality of CHM.

>Usually specific help browsers have some benefits over normal web browsers like extended search facilities. As I am no Windows user, I have no experience and no strong opinion. As the currently used browser is no longer supported by MS, what are they offering instead?

About Windows, my opinion is the same as Microsoft. 
If you install gnucash-docs on the localhost, Windows index search will be used.
If you put gnucash-docs on the internet, search engines like google will be used.
Microsoft lead us to follow the guideline and they plan to remove the local help browser such as htmlhelp.exe and CHM, completely.

Reply:
>About Windows, my opinion is the same as Microsoft.
>If you install gnucash-docs on the localhost, Windows index search will be used.
> If you put gnucash-docs on the internet, search engines like google will be used.

I think these MDN pages are our goal.
https://developer.mozilla.org/ja/docs/Learn/Getting_started_with_the_web/CSS_basics

Reply:
> Windows code pages are not compliant with ISO codes like ISO-8859-*. Moreover, the multibyte charset is more complex. Windows-1252, CP1252, MSCP1252, and WINDOWS-1252 are not suitable for iexplore.exe because they are not defined by IANA.

Why are you referring to iexplore here ? I thought chm was using a custom help viewer. The change only applies to the chm format which is as you write yourself old and not utf-8 capable (which IE is by the way) so it seems to me iexplore  is irrelevant here ?

Other than that I have no idea whether there's a problem here or not and how to potentially fix it.

Reply:
Changing to html is probably a reasonable choice, though that needs coordinated changes in gnucash-on-windows and  gnucash. And testing to check if it all works as we want it to.